### PR TITLE
PUBDEV-7006: Added missing parameters noise and seed to R new TE API

### PR DESCRIPTION
**Pull request by deil87**
This PR is only dedicated to a fix in R.

Agenda:
`h2o.transform` method for H2OTargetEncoderModel in R was missing `noise` parameter and also `seed` parameter.

### CODE REVIEW

File: h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R [hunk 1]
```diff
@@ -1,6 +1,10 @@
 setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
 source("../../../scripts/h2o-r-test-setup.R")
 
+doTestAndContinue <- function(testDesc, test) {
```
Comment (michalkurka) [inline h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R]: Please use the already existing method `doTest`. This is copy-paste-modify. Not maintainable.

File: h2o-r/h2o-package/R/models.R [hunk 2]
```diff
@@ -514,7 +514,9 @@ setMethod("h2o.transform", signature("H2OTargetEncoderModel"), function(model, d
                                                                         data_leakage_handling = NULL,
                                                                         use_blending = NULL,
                                                                         inflection_point = -1,
-                                                                        smoothing = -1) {
+                                                                        smoothing = -1,
```
Comment (michalkurka) [inline h2o-r/h2o-package/R/models.R]: Please document the added parameters.

File: h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R (hunk 1 above)
Comment (deil87) [reply h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R]: @michalkurka sure, I'm aware of `doTest` method. Btw I wonder why we are ok with for example `get_partitioned_model_names` method in python then? But I agree that it is better to have utils that are being shared between all tests. The reason why I added this modification is that I don't wan't one-file-per-test approach. And I don't want to put all use cases into one test like we usually do in R. Maybe we should agree upon some util method and add it to test utils?

File: h2o-r/h2o-package/R/models.R (hunk 2 above)
Comment (deil87) [reply h2o-r/h2o-package/R/models.R]: @michalkurka Thanks! I've missed that.

File: h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R (hunk 1 above)
Comment (deil87) [reply h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R]: I've found `doSuite` in utils so I think it will work for me.

**Comment by deil87** [comment]
@michalkurka tests look fine. Going to merge this one today.

**Pull request merged**